src/key_exchange/group/p256.rs

```diff
@@ -9,39 +9,39 @@ use super::KeGroup;
 use crate::errors::InternalError;
 use generic_array::typenum::{U32, U33};
 use generic_array::GenericArray;
+use p256_::elliptic_curve::group::GroupEncoding;
+use p256_::elliptic_curve::sec1::ToEncodedPoint;
+use p256_::elliptic_curve::{PublicKey, SecretKey};
+use p256_::NistP256;
 use rand::{CryptoRng, RngCore};
 
-impl KeGroup for p256_::ProjectivePoint {
+impl KeGroup for PublicKey<NistP256> {
     type PkLen = U33;
     type SkLen = U32;
 
     fn from_pk_slice(element_bits: &GenericArray<u8, Self::PkLen>) -> Result<Self, InternalError> {
-        use p256_::elliptic_curve::group::GroupEncoding;
-
-        Option::from(Self::from_bytes(element_bits)).ok_or(InternalError::PointError)
+        Self::from_sec1_bytes(element_bits).map_err(|_| InternalError::PointError)
     }
 
     fn random_sk<R: RngCore + CryptoRng>(rng: &mut R) -> GenericArray<u8, Self::SkLen> {
-        use p256_::elliptic_curve::Field;
-
-        p256_::Scalar::random(rng).into()
+        SecretKey::<NistP256>::random(rng).to_bytes()
     }
 
     fn public_key(sk: &GenericArray<u8, Self::SkLen>) -> Self {
-        Self::generator() * p256_::Scalar::from_bytes_reduced(sk)
+        SecretKey::<NistP256>::from_bytes(sk).unwrap().public_key()
```

Comment:
These `unwrap`s have to be removed.

My idea was to use the appropriate `SecretKey` type instead of storing `GenericArray`s everywhere. I believe this would be better anyway, because conversion isn't free (it's very cheap though) and any checking is now guaranteed in the type and can now be handled during serialization.